src/metabase/channel/shared.clj

```diff
@@ -25,3 +31,88 @@
   [part]
   (when part
     (m/update-existing-in part [:result :data :rows] maybe-deref)))
+
+(defn- schedule-timezone
+  []
+  (or (driver/report-timezone) "UTC"))
+
+(defn- format-time
+  "Format hour and minute into a 12-hour time string with AM/PM."
+  [hour minute]
+  (let [hour-12 (cond
+                  (zero? hour) 12
+                  (> hour 12) (- hour 12)
+                  :else hour)
+        am-pm (if (< hour 12) "AM" "PM")]
+    (if-not (zero? minute)
+      (format "%d:%02d %s" hour-12 minute am-pm)
+      (format "%d %s" hour-12 am-pm))))
+
+(defn- cron-description
+  [cron-string]
+  (try
+    (let [s (CronExpressionDescriptor/getDescription ^String cron-string
+                                                     (doto (Options.)
+                                                       (.setZeroBasedDayOfWeek false)))
+          s (str (u/lower-case-en (subs s 0 1))
+                 (subs s 1))]
+      (format "Run %s %s" s (schedule-timezone)))
+    (catch Exception e
+      (log/errorf e "Failed to parse cron expression: %s" cron-string)
+      nil)))
+
+(defn friendly-cron-description
+  "Convert a cron string to a human-readable description."
+  [cron-string]
+  (let [[seconds minutes hours day-of-month month day-of-week _] (str/split cron-string #"\s+")
```

Comment:
nit: Might was well call the last arg `_year` in case we ever need it / so its clearer why we're not using it.

I can't imagine we'd ever let it be configured, but might as well check that it's nil or `*` for all our pattern matches?

Reply:
good point, added that check and also add tests for 7 components